agent/command/git.go

```diff
@@ -866,6 +866,10 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 
 	// For every module, expand the module prefix.
```

Comment:
Hm that makes sense. I thought about what might differentiate this between other build variant fields and I think it's that the modules section is a bit of an anti-pattern.

We've seen this with the correctness ticket, that the git.get_project attempts to clone more repo's than is listed in the build variant definition (since they re-used the same func/git.get_project across all build_variant's). The modules section is _only_ tied to the git.get_project command while every other field ties in to most/every stage of a task. 

I think in an ideal world, build variants wouldn't list the modules and the git.get_project's would only. How it's set up now, the modules section of the build variant are really just param's for git.get_project. 

I'm okay if we do push back on this, since it is something that we generally can't support with any other build variant fields, but I do think it's different enough (and if we wanted to, could make a mini-project of migrating the fields) to warrant an exception. I could increase the wording in the docs / our code to exemplify that.

Reply:
I think you're right that modules are something of an exception since we allow expansions for that section already; I do think we should clarify that this is an exception in the docs so I'll leave a comment to clarify.